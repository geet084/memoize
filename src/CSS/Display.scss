.display {
  display: flex;
  justify-content: center;
}

.intro {
  color: #fff;
  width: 450px;
  text-align: center;
  background-color: #2D4C46;
  padding: 40px;
  opacity: .8;
  font-size: 3rem;
  border-radius: 15px;
}