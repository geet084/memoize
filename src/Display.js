import React, { Component } from 'react';
import './CSS/Main.scss';
import Card from './Card.js'

export default class Display extends Component {
  constructor() {
    super();

    this.state = {
      correctlyAnswered: null
    }
  }

  render() {
    return (
      <section>
        <Card questions={this.props.prototypes}
          answer={this.props.answer}
          answered={this.props.answered} />
      </section>
    )
  }
}