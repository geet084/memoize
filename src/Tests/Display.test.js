import React from 'react';
import Display from '../Display.js';
import { shallow } from 'enzyme';
import prototypes from '../datasets/aType.js'

const mockData = prototypes;
const mockAnswer = jest.fn();
const mockAnsweredQuestions = jest.fn();


describe('Display', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Display prototypes={mockData}
               currentAnswer={mockAnswer}
               answeredQuestions={mockAnsweredQuestions} />
    )
  })

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should have a proper default state', () => {
    expect(wrapper.state()).toEqual({
      correctlyAnswered: null
    })
  })

});

