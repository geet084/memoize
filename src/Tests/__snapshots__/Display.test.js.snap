// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display should match the snapshot with all data passed in correctly 1`] = `
<section
  className="display"
>
  <Card
    answeredQuestions={[MockFunction]}
    questions={
      Object {
        "aTypes": Array [
          Object {
            "concat": "This method is used to merge two or more arrays. This method also does not change the existing arrays, but instead returns a new array.",
          },
          Object {
            "copyWithin": "This method shallow copies part of an array to another location in the same array and returns it, without modifying its size.",
          },
          Object {
            "entries": "This method returns a new Array Iterator object that contains the key/value pairs for each index in the array.",
          },
          Object {
            "every": "This method tests whether all elements in the array pass the test implemented by the provided function.",
          },
          Object {
            "fill": "This method fills all the elements of an array from a start index to an end index with a static value. The end index is not included.",
          },
          Object {
            "filter": "This method creates a new array with all elements that pass the test implemented by the provided function.",
          },
          Object {
            "find": "This method returns the value of the first element in the array that satisfies the provided testing function. Otherwise undefined is returned.",
          },
          Object {
            "findIndex": "This method returns the index of the first element in the array that satisfies the provided testing function. Otherwise, it returns -1, indicating no element passed the test.",
          },
          Object {
            "flat": "This method creates a new array with all sub-array elements concatenated into it recursively up to the specified depth.",
          },
          Object {
            "flatMap": "This method first maps each element using a mapping function, then flattens the result into a new array. It is identical to a map followed by a flat of depth 1, but flatMap is often quite useful, as merging both into one method is slightly more efficient.",
          },
          Object {
            "forEach": "This method executes a provided function once for each array element.",
          },
          Object {
            "includes": "This method determines whether an array includes a certain value among its entries, returning true or false as appropriate.",
          },
          Object {
            "indexOf": "This method returns the first index at which a given element can be found in the array, or -1 if it is not present.",
          },
          Object {
            "join": "This method creates and returns a new string by concatenating all of the elements in an array (or an array-like object), separated by commas or a specified separator string.",
          },
          Object {
            "keys": "This method returns a new Array Iterator object that contains the keys for each index in the array.",
          },
          Object {
            "lastIndexOf": "This method returns the last index at which a given element can be found in the array, or -1 if it is not present. The array is searched backwards, starting at fromIndex.",
          },
          Object {
            "map": "This method creates a new array with the results of calling a provided function on every element in the calling array.",
          },
          Object {
            "pop": "This method removes the last element from an array and returns that element. This method also changes the length of the array.",
          },
          Object {
            "push": "This method adds one or more elements to the end of an array and returns the new length of the array.",
          },
          Object {
            "reduce": "This method executes a reducer function (that you provide) on each member of the array resulting in a single output value.",
          },
          Object {
            "reduceRight": "This method applies a function against an accumulator and each value of the array (from right-to-left) to reduce it to a single value.",
          },
          Object {
            "reverse": "This method reverses an array in place. The first array element becomes the last, and the last array element becomes the first.",
          },
          Object {
            "shift": "This method removes the first element from an array and returns that removed element. This method also changes the length of the array.",
          },
          Object {
            "slice": "This method returns a shallow copy of a portion of an array into a new array object selected from begin to end (end not included). The original array will not be modified.",
          },
          Object {
            "some": "This method tests whether at least one element in the array passes the test implemented by the provided function.",
          },
          Object {
            "sort": "This method sorts the elements of an array in place and returns the array. The default sort order is built upon converting the elements into strings, then comparing their sequences of UTF-16 code units values.",
          },
          Object {
            "splice": "This method changes the contents of an array by removing or replacing existing elements and/or adding new elements.",
          },
          Object {
            "toLocaleString": "This method returns a string representing the elements of the array. The elements are converted to Strings using their toLocaleString methods and these Strings are separated by a locale-specific String (such as a comma “,”).",
          },
          Object {
            "toString": "This method returns a string representing the specified array and its elements.",
          },
          Object {
            "unshift": "This method adds one or more elements to the beginning of an array and returns the new length of the array.",
          },
          Object {
            "values": "This method returns a new Array Iterator object that contains the values for each index in the array.",
          },
        ],
      }
    }
  />
</section>
`;
